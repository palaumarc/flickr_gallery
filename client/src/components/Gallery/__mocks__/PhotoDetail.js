import React from "react";

const PhotoDetail = props => <photo-detail-mock {...props}>{props.children}</photo-detail-mock>;

export default PhotoDetail;