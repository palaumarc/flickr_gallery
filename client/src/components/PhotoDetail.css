/* Modal background */
.photo-detail {
    position: fixed;
    z-index: 1;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    overflow: auto;
    background-color: rgba(70, 70, 70, 0.7);
}

/* Modal content (layout) */
.photo-detail-content {
    display: grid;
    padding-left: 5px;
    padding-right: 5px;
    height: 100%;
    grid-template-rows: 30px auto 20% 10%;
    grid-template-columns: 100%;
    grid-gap: 10px;
    justify-items: center;
    align-items: start;
}

/* Modal image */
.photo-detail-content img {
    max-width: 100%;
    max-height: 100%;
    align-self: center;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

/* Photo info container */
.photo-detail-content-info-container {
    color: white;
    padding: 10px;
    width: 90%;
    max-width: 800px;
    height: 100%;
    overflow-y: auto;
}

.photo-detail-content-info-container p {
    font-weight: lighter;
    word-break: break-word;
}

.photo-detail-content-info-username {
    font-style: italic;
    cursor: pointer;
    word-break: break-word;
}

.photo-detail-content-info-username:hover {
    color: rgb(212, 212, 212);
}

.photo-detail-content-info-title {
    font-weight: bold;
    word-break: break-word;
}

/* Modal Close Button */
.photo-detail-close {
    color: white;
    font-size: 35px;
    font-weight: bold;
    cursor: pointer;
    z-index: 2;
    justify-self: end;
}

.photo-detail-close:hover,
.photo-detail-close:focus {
    color: rgb(212, 212, 212);
    text-decoration: none;
}

/* Modal next and previous photo control */
.photo-detail-content-next-previous-container {
    width: 100px;
    align-self: center;
}

.photo-detail-content-next, .photo-detail-content-previous {
    color: white;
    font-size: 20px;
    font-weight: bold;
    cursor: pointer;
    z-index: 2;
}

.photo-detail-content-next {
    float: right;
}

.photo-detail-content-previous {
    float: left;
}

.photo-detail-content-next:hover, .photo-detail-content-previous:hover { 
    color: rgb(212, 212, 212);
}