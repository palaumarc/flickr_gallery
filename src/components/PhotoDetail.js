import React from 'react';

export default PhotoDetail = ({ photo, onClose }) => (
    <div>
        <div onClick={onClose}>Close</div>
        <img src={photo.fullSizeUrl} alt={photo.title}/>
    </div>
);