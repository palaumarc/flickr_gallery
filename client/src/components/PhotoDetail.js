import React from 'react';
import './PhotoDetail.css'

const PhotoDetail = ({ photo, onClose, previousControl, nextControl }) => (
    <div className="photo-detail">
        <div className="photo-detail-content">
            <div className="photo-detail-close" onClick={onClose}>&times;</div>
            <img src={photo.fullSizeUrl || photo.previewUrl} alt={photo.title} />
            <div className="photo-detail-content-info-container">
                <span className="photo-detail-content-info-title">{photo.title}</span> - <span className="photo-detail-content-info-username" onClick={() => window.open(photo.flickrRedirectUrl)}>{photo.username}</span>
                <p>{photo.description}</p>
            </div>
            <div className="photo-detail-content-next-previous-container">
                {previousControl}
                {nextControl}
            </div>
        </div>
    </div>
);

PhotoDetail.PreviousPhotoArrow = ({onClick}) => (
    <div className="photo-detail-content-previous" onClick={onClick}>&lt;</div>
);

PhotoDetail.NextPhotoArrow = ({onClick}) => (
    <div className="photo-detail-content-next" onClick={onClick}>&gt;</div>
);

export default PhotoDetail;