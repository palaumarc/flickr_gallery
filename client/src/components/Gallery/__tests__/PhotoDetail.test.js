import React from "react";
import renderer from "react-test-renderer";
import PhotoDetail from "../PhotoDetail";

const photoMock = {
    previewUrl: 'previewUrl',
    fullSizeUrl: 'fullSizeUrl',
    title: 'title',
    flickrRedirectUrl: 'flickrRedirectUrl',
    username: 'username',
    description: 'description'
}

test('Snapshot with a photo object with fullSizeUrl', () => {
    const tree = renderer.create(<PhotoDetail photo={photoMock}/>);
    expect(tree.toJSON()).toMatchSnapshot();
})

test('Snapshot with a photo object without fullSizeUrl', () => {
    const photoMockWithoutFullSizeUrl = {...photoMock, fullSizeUrl: undefined}
    const tree = renderer.create(<PhotoDetail photo={photoMockWithoutFullSizeUrl}/>);
    expect(tree.toJSON()).toMatchSnapshot();
})