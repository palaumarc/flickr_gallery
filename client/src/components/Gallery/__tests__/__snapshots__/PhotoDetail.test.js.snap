// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot with a photo object with fullSizeUrl 1`] = `
<div
  className="photo-detail-content"
>
  <img
    alt="title"
    src="fullSizeUrl"
  />
  <div
    className="photo-detail-content-info-container"
  >
    <span
      className="photo-detail-content-info-title"
    >
      title
    </span>
    -
    <span
      className="photo-detail-content-info-username"
      onClick={[Function]}
    >
      username
    </span>
    <br />
    <span
      className="photo-detail-content-info-description"
    >
      description
    </span>
  </div>
</div>
`;

exports[`Snapshot with a photo object without fullSizeUrl 1`] = `
<div
  className="photo-detail-content"
>
  <img
    alt="title"
    src="previewUrl"
  />
  <div
    className="photo-detail-content-info-container"
  >
    <span
      className="photo-detail-content-info-title"
    >
      title
    </span>
    -
    <span
      className="photo-detail-content-info-username"
      onClick={[Function]}
    >
      username
    </span>
    <br />
    <span
      className="photo-detail-content-info-description"
    >
      description
    </span>
  </div>
</div>
`;
